### PR TITLE
Add `mockOnly` option to `@Instantiable` to support hand-written mock providers

### PR DESCRIPTION
**Pull request by dfed**
### Motivation
- Introduce a `mockOnly` declaration mode for `@Instantiable` so types can participate only in mock generation via a hand-written `public static func mock(...)` without requiring an `init`/`instantiate()` or `Instantiable` conformance.
- Enforce invariants and surface helpful diagnostics when `mockOnly` is misused (e.g. combined with `generateMock` or `isRoot`, or missing a `mock()` method).

### Description
- Add a `mockOnly: Bool = false` parameter to the `@Instantiable` macro and expose it via `AttributeSyntax` helpers (`mockOnlyValue`).
- Extend the `Instantiable` model with a `mockOnly` property and thread this through parsing in `InstantiableVisitor` and normalization in `GenerateCommand`.
- Update the macro implementation in `InstantiableMacro` to validate `mockOnly` semantics, allow skipping enforced `Instantiable` conformance/initializer requirements for `mockOnly` declarations, and emit new diagnostic errors when `mockOnly` is combined with `generateMock` or `isRoot`, or when the required hand-written `mock()` is missing.
- Modify the dependency tree and mock-generation logic to accept `allInstantiables` and prefer hand-written mock providers when computing mock scopes, and to ignore `mockOnly` declarations when mapping the primary fulfilling instantiables while still counting mock providers to detect duplicates.
- Add new diagnostic cases and fix-its to `FixableInstantiableError` for `mockOnly` misuse and update corresponding messages.
- Update documentation in `Documentation/Manual.md` to describe `mockOnly` usage.
- Add and update unit tests covering `mockOnly` behavior, diagnostics, mock-preference logic, and duplicate mock provider detection across decorations.

### Testing
- Updated macro tests in `Tests/SafeDIMacrosTests/InstantiableMacroTests.swift` were run and verify macro expansion and diagnostics for `mockOnly` scenarios and they passed.
- Updated `FixableInstantiableError` unit tests in `Tests/SafeDICoreTests/FixableInstantiableErrorTests.swift` were run and passed.
- New integration tests in `Tests/SafeDIToolTests` exercising mock-generation behavior with `mockOnly` types and duplicate mock-provider errors were executed and passed.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_b_69daa5e691888323ac031e7bc62028dc)

### CODE REVIEW

File: Sources/SafeDICore/Models/InstantiableStruct.swift [hunk 1]
```diff
@@ -68,6 +70,8 @@ public struct Instantiable: Codable, Hashable, Sendable {
 	public let mockAttributes: String
 	/// Whether to generate a `mock()` method for this type.
 	public let generateMock: Bool
+	/// Whether this declaration is used only for mock generation.
+	public let mockOnly: Bool
```
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Models/InstantiableStruct.swift:74]: Builds are hermetic. It's fine
